App.tsx

```diff
@@ -63,6 +64,7 @@ const App = () => {
     <StoreProvider>
       <AgentProvider agent={agent}>
         <StatusBar barStyle="light-content" hidden={false} backgroundColor={Colors.primary} translucent={false} />
+        <ConnectionModal />
```

Comment:
I'm generally liking the concept. Wondering if we could/would do the same for a Credential Offer or Proof Request? Perhaps not ATM since they are handled from the Home Screen

Reply:
To be honest this seems to create better separation than having that modal logic baked into the scan screen and would be pretty useful if we use different connection types. 👍

Reply:
I think so. Most events are notably async so probably better to get some sort of background service dealing with that and the UI just focused on presentation when the state changes.